wagtail/test/testapp/models.py

```diff
@@ -1967,6 +1967,22 @@ class FileGenericSetting(BaseGenericSetting):
     file = models.FileField()
 
 
+@register_setting
+class PreviewableSiteSetting(PreviewableMixin, BaseSiteSetting):
+    text = models.TextField()
+
+    def get_preview_template(self, request, mode_name):
+        return "tests/previewable_setting.html"
+
+
+@register_setting
+class PreviewableGenericSetting(PreviewableMixin, BaseGenericSetting):
+    text = models.TextField()
+
+    def get_preview_template(self, request, mode_name):
+        return "tests/previewable_setting.html"
```

Comment:
I know I said one test model was enough, but after looking at the code I think we need at least one generic setting and one site setting to cover the whole preview view. I added `PreviewableSiteSetting` and made the tests to run with both models.